tensorboard/data/server/event_file.rs

```diff
@@ -68,14 +70,30 @@ impl<R: Read> EventFileReader<R> {
         Self {
             last_wall_time: None,
             reader: TfRecordReader::new(reader),
+            checksum: true,
         }
     }
 
+    /// Sets whether to compute checksums for records before parsing them as protos.
+    pub fn checksum(&mut self, yes: bool) {
+        self.checksum = yes;
+    }
+
     /// Reads the next event from the file.
     pub fn read_event(&mut self) -> Result<Event, ReadEventError> {
         let record = self.reader.read_record()?;
-        record.checksum()?;
-        let event = Event::decode(&record.data[..])?;
+        let event = if self.checksum {
+            record.checksum()?;
+            Event::decode(&record.data[..])?
+        } else {
+            match Event::decode(&record.data[..]) {
+                Ok(proto) => proto,
+                Err(e) => {
```

Comment:
Nice. Checking checksum on error!

Reply:
Yep, I figured that it could come in handy, and it’s easy enough and
costs nothing in the happy path.